builds/e2e/templates/nested-longhaul-deploy.yaml

```diff
@@ -0,0 +1,166 @@
+      CleanTargetFolder: true
+  - task: CopyFiles@2
+    displayName: 'Copy aziot-identity-service'
+    inputs:
+      SourceFolder: "$(Build.StagingDirectory)"
+      Contents: "$(aziotis.package.filter)"
+      TargetFolder: "$(Agent.HomeDirectory)/../artifacts/"
+  - task: Bash@3
+    displayName: 'Generate device certificates'
+    inputs:
+      targetType: inline
+      workingDirectory: "/certs"
+      script: |
+        sudo chmod -R 777 /certs
+        deviceId="${{ parameters['release.label'] }}-${{ parameters['metricsCollector.hostPlatform'] }}-longhaul-${{ parameters['test.buildId'] }}"
+        echo "Installing test certificate bundle for edge device id ${deviceId}"
+        ./certGen.sh create_edge_device_certificate "${deviceId}"
+        
+        echo "##vso[task.setvariable variable=TrustBundle]/certs/certs/azure-iot-test-only.root.ca.cert.pem"
+        echo "##vso[task.setvariable variable=DeviceCaCert]/certs/certs/iot-edge-device-${deviceId}-full-chain.cert.pem"
+        echo "##vso[task.setvariable variable=DeviceCaKey]/certs/private/iot-edge-device-${deviceId}.key.pem"
+  - task: Bash@3
+    displayName: 'Run Long Haul Deployment'
+    inputs:
+      targetType: inline
+      script: |
+        declare -a cnreg=( ${{ parameters['container.registry.credential'] }} )
+        . $(Agent.HomeDirectory)/../artifacts/${{ parameters['images.artifact.name'] }}/artifactInfo.txt
+        testName="LongHaul"
+        
+        testInfo="TestBuildNumber=${{ parameters['test.buildNumber'] }}"
+        testInfo="$testInfo,TestBuildId=${{ parameters['test.buildId'] }}"
+        testInfo="$testInfo,BuildSourceBranch=${{ parameters['build.source.branch'] }}"
+        testInfo="$testInfo,EdgeletSourceBranch=${{ parameters['edgelet.source.branch'] }}"
+        testInfo="$testInfo,ImagesSourceBranch=${{ parameters['images.source.branch'] }}"
+        testInfo="$testInfo,HostPlatform=${{ parameters['metricsCollector.hostPlatform'] }}"
+        testInfo="$testInfo,TestName=$testName"
+        
+        chmod +x $(Build.SourcesDirectory)/scripts/linux/trcE2ETest.sh
+        
+        sudo --preserve-env $(Build.SourcesDirectory)/scripts/linux/trcE2ETest.sh \
+            -testDir "$(Agent.HomeDirectory)/.." \
+            -releaseLabel "${{ parameters['release.label'] }}" \
+            -artifactImageBuildNumber "$BuildNumber" \
+            -testName "$testName" \
+            -testMode "$testName" \
+            -containerRegistry "${{ parameters['container.registry'] }}" \
+            -containerRegistryUsername "${{ parameters['container.registry.username'] }}" \
+            -containerRegistryPassword "${{ parameters['container.registry.password'] }}" \
+            -iotHubConnectionString "${{ parameters['iotHub.connectionString'] }}" \
+            -eventHubConnectionString "${{ parameters['eventHub.connectionString'] }}" \
+            -testDuration  24:00:00 \
```

Comment:
Umm... what? I don't think we want to run "Long Haul" for 24hr. 
Please make this a configurable parameter from pipeline.

Reply:
fixed